Substitute this with self

hubertp opened the pull request:
### Pull Request Description

A semi-manual s/this/self appied to the whole standard library.
Related to https://www.pivotaltracker.com/story/show/182328601

In the compiler promoted to use constants instead of hardcoded
`this`/`self` whenever possible.

### Important Notes

The PR **does not** require explicit `self` parameter declaration for methods as this part 
of the design is still under consideration.

### Checklist

Please include the following checklist in your PR:

- [ ] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide dist` and `./run ide watch`.

hubertp commented:
I dare anyone to review `distribution/lib`. I double dare you ;)

radeusgd commented:
> I dare anyone to review `distribution/lib`. I double dare you ;)

I've got a long train ride tomorrow evening, so challenge accepted ^^

hubertp committed:
Changelog.md

hubertp committed:
nits

hubertp committed:
Fix tests

radeusgd commented on distribution/lib/Standard/Base/0.0.0-dev/src/Data/Json.enso:
```diff
@@ -183,7 +183,7 @@ type No_Such_Field_Error field_name
 No_Such_Field_Error.to_display_text : Text
 No_Such_Field_Error.to_display_text =
-    "The field " + this.field_name.to_text + " is not present in this object."
+    "The field " + self.field_name.to_text + " is not present in self object."
```
```suggestion
    "The field " + self.field_name.to_text + " is not present in this object."
```

radeusgd commented on distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso:
```diff
@@ -276,13 +276,13 @@ type Number
        Arguments:
        - that: The number to check equality against.
-       - epsilon: The value by which `this` and `that` can be separated by before
+       - epsilon: The value by which `self` and `that` can be separated by before
```
I wonder what we should do about these?

`this` and `that` sounded nicely, now, not sure if we should change `that` to something else? But I'm not sure if there are good alternatives either.

hubertp replied:
Yes, this is something @ekmett raised as well

radeusgd commented on distribution/lib/Standard/Base/0.0.0-dev/src/System/File/Option.enso:
```diff
@@ -45,9 +45,9 @@ type Option
     ## PRIVATE
 
-       Convert this object into a representation understandable by the JVM.
+       Convert self object into a representation understandable by the JVM.
```
```suggestion
       Convert this object into a representation understandable by the JVM.
```

radeusgd commented on distribution/lib/Standard/Database/0.0.0-dev/src/Data/Column.enso:
```diff
@@ -357,11 +357,11 @@ type Column
-       `other` and each element of `this`.  If `other` is a column, the
-       operation is performed pairwise between corresponding elements of `this` ]
+       `other` and each element of `self`.  If `other` is a column, the
+       operation is performed pairwise between corresponding elements of `self` ]
```
```suggestion
       operation is performed pairwise between corresponding elements of `self`
```

radeusgd commented on lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/model/MethodDefinition.java:
```diff
@@ -198,7 +198,7 @@ public static class ArgumentDefinition {
     private static final String CALLER_INFO = "org.enso.interpreter.runtime.callable.CallerInfo";
     private static final String DATAFLOW_ERROR = "org.enso.interpreter.runtime.error.DataflowError";
-    private static final String THIS = "this";
+    private static final String THIS = "self";
```
Don't we want to rename `THIS` to `SELF` too?

hubertp replied:
It doesn't hurt, yes, although this is Java-specific

radeusgd replied:
Ah, in that case maybe indeed it's better to keep it as-is.

hubertp replied:
I think @jdunkerley kind of expected for that to happen during this PR so I will make that change

radeusgd reviewed (approved):
I've skimmed through all the changes and apart from the few comments above it looks good to me.

I'd like to discuss the `that` issue, but it likely can be addressed separately from this PR, as it is already huge as it is.

hubertp committed:
Apply suggestions from code review

Co-authored-by: Radosław Waśko <radoslaw.wasko@enso.org>

JaroslavTulach reviewed (approved):
So many changed files! Let's get this done.

hubertp committed:
More fun with _this -> self

hubertp committed:
Substitute this with self

A semi-manual s/this/self appied to the whole standard library.
Related to https://www.pivotaltracker.com/story/show/182328601

In the compiler promoted to use constants instead of hardcoded
`this`/`self` whenever possible.

4e6 reviewed (approved):
Epic!

hubertp committed:
Adapt micro distribution

mergify[bot] committed:
Merge branch 'develop' into wip/hubert/this-to-self-182328601

hubertp committed:
Minimal GUI changes

hubertp committed:
Missed some this after rebase

mergify[bot] committed:
Merge branch 'develop' into wip/hubert/this-to-self-182328601

hubertp commented:
@MichaelMauderer @mwu-tow @wdanilo @farmaazon please ignore everything apart from `api/gui` stuff. No need for you to waste time on this monster PR.

hubertp committed:
Removing remaining 'this' usage from gui

mergify[bot] committed:
Merge branch 'develop' into wip/hubert/this-to-self-182328601
